src/client/api/event.c

```diff
@@ -478,16 +478,20 @@ daos_event_complete(struct daos_event *ev, int rc)
 		D_MUTEX_LOCK(&eqx->eqx_lock);
 	}
 
-	D_ASSERT(evx->evx_status == DAOS_EVS_RUNNING ||
-		 evx->evx_status == DAOS_EVS_ABORTED);
+	if (evx->evx_status == DAOS_EVS_READY ||
+	    evx->evx_status == DAOS_EVS_COMPLETED ||
+	    evx->evx_status == DAOS_EVS_ABORTED)
```

Comment:
any reasonable case to call daos_event_complete() for event with DAOS_EVS_READY or DAOS_EVS_COMPLETED status?

Reply:
im not sure, but it shouldn't really be an assertion, and i thought we could just do nothing there.
do you prefer to return error there?